### PR TITLE
BUG: Series.at returning Series with one element instead of scalar

### LINKED ISSUE #38053: BUG: calling at[] on a Series with a single-level MultiIndex returns a Series, not a scalar
- [YES] I have checked that this issue has not already been reported.

- [YES] I have confirmed this bug exists on the latest version of pandas.

- [NO] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
idx1 = pd.MultiIndex.from_tuples([(False, 1.0), 
                                  (False, 10.0),
                                  (True, 1.0), 
                                  (True, 10.0)])
s1 = pd.Series((0, 1, 2, 3), index = idx1)
s1.at[(False, 1.0)]   # returns '0', expected output

idx2 = pd.MultiIndex.from_tuples([(False,), (True,)])
s2 = pd.Series((0, 1), index = idx2)
s2.at[False]  # returns a pandas.Series with one element

```

#### Problem description

As per the user guide, I expect `at[]` to return a scalar. However, when I index a `Series` with a `MultiIndex` with only one level, `at[]` returns a `Series` with a single element. A `Series` with a multi-level `MultiIndex` works as expected.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.6.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-54-generic
Version          : #60-Ubuntu SMP Fri Nov 6 10:37:59 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.4
setuptools       : 50.3.1.post20201107
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : None
bottleneck       : 1.3.2
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #38053
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I think if we are at this place and len(values) is 1, we want to return a scalar instead of a Series. This happens only because loc is a slice (0,1) with one MultiIndex level instead of a scalar.

Should I create a file to move all MultiIndex at tests to the MultiIndex folder?

### CODE REVIEW

File: pandas/tests/indexing/test_scalar.py [hunk 1]
```diff
@@ -300,3 +300,10 @@ def test_multiindex_at_set():
     assert series.at[1, 3] == 5
     series.loc[1, 3] = 6
     assert series.loc[1, 3] == 6
+
+
+def test_at_get_multiindex_one_level():
+    # GH#38053
+    s2 = Series((0, 1), index=[[False, True]])
```
Comment (jreback) [inline pandas/tests/indexing/test_scalar.py:307]: I think location is fine, but can you group the MultiIndex tests into a class (in this file).

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_scalar.py [hunk 2]
```diff
@@ -300,3 +300,10 @@ def test_multiindex_at_set():
     assert series.at[1, 3] == 5
     series.loc[1, 3] = 6
     assert series.loc[1, 3] == 6
+
+
+def test_at_get_multiindex_one_level():
```
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/tests/indexing/test_scalar.py:305]: In the same file above and test_at. One in test_partial, but this one is checking different things

Renamed the test

File: pandas/tests/indexing/test_scalar.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/test_scalar.py:307]: Done

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -625,6 +625,7 @@ Indexing
 - Bug in :meth:`DataFrame.iloc` and :meth:`Series.iloc` aligning objects in ``__setitem__`` (:issue:`22046`)
 - Bug in :meth:`DataFrame.loc` did not raise ``KeyError`` when missing combination was given with ``slice(None)`` for remaining levels (:issue:`19556`)
 - Bug in :meth:`DataFrame.loc` raising ``TypeError`` when non-integer slice was given to select values from :class:`MultiIndex` (:issue:`25165`, :issue:`24263`)
+- Bug in :meth:`Series.at` returning :class:`Series` with one element instead of scalar (:issue:`38053`)
```
[the remark this answers is not in the document]
Comment (phofl) [reply doc/source/whatsnew/v1.2.0.rst]: Improved the message, thx

**Comment by jreback** [comment]
hmm, lots of things are failing, can you merge master and ping on green.

**Comment by phofl** [comment]
Good test, caught a bug. Adjusted the code.

**Review by jreback — APPROVED** [review]

**Comment by phofl** [comment]
@jreback green now

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**